### ISSUE TITLE
Logged-out visibility toggle not applying to API responses

### ISSUE THREAD

**Issue by pseudo-security** [bluesky-social/social-app#2909]
**Describe the bug**
This is related / similar to #2574 . If you have the Logged-out visibility toggle on, the bsky.app web version is still returning the plain-text content in the API response, even if the UI is hiding it;

**To Reproduce**

1. Toggle Logged-out visibility on
2. Check your own post in an incognito tab

**Expected behavior**

All post content should be hidden, and not returned/visible in API responses.

**Screenshots**

"Sign-in Required" post returning content in API response:
![firefox_afmyRXHa6u](https://github.com/bluesky-social/social-app/assets/59893409/b844d503-e014-4751-b567-49741146c19d)

"Sign-in Required" profile returning profile description in API response:
![firefox_RINrHnpbht](https://github.com/bluesky-social/social-app/assets/59893409/d3f7a010-4357-4871-8ff2-dae3a3617ea9)

**Details**

 - Platform: Web
 - Platform version: 🤷 whatever is in prod as of 2024-02-17

**Comment by surfdude29** [comment]
As I understand it, the logged-out visibility toggle is only designed to add a small amount of friction to accessing the content of a user who has it enabled. It does not prevent the content from being available via the API, it only affects the display of the content in the Bluesky web interface and in the Bluesky app.

Thus the whole point of the feature is to hide the content in the UI, as is now correctly being done; the data returned in API responses is unaffected.

**Comment by pseudo-security** [comment]
> Bluesky will not show your profile and posts to logged-out users. Other apps may not honor this request. This does not make your account private.
> 
> Note: Bluesky is an open and public network. This setting **_only limits the visibility of your content on the Bluesky app and website_** [emphasis mine], and other apps may not respect this setting. Your content may still be shown to logged-out users by other apps and websites.
The content is visible to me while not logged into the Bluesky website, I am able to read the content and replies (quoted or not). Bluesky is essentially accessing the raw responses of the unfiltered API protocol server, and then interpreting it client-side. BS has the ability to apply appropriate mechanisms server-side for their specific client-server interactions, without affecting how 3rd-party federating services receive the content. The broader problem is BS has introduced a privacy-setting, implemented it via client-side mechanisms, and has not plumbed it through the server implementation, nor the spec. I think the feature should either be removed, or fixed on BS server (assuming the AT protocol will not add this content control).

**Comment by surfdude29** [comment]
> The content is visible to me while not logged into the Bluesky website

I think that's a highly debatable statement. If you can only see the public content that we're talking about by looking in developer tools at the API responses, then it's not visible on the website. (If we were talking about private content in another context it would be a different situation.)

> The broader problem is BS has introduced a privacy-setting

The logged-out visibility toggle is not a privacy setting. It appears in the Moderation tab and the text states explicitly: "This does not make your account private."

The bottom line is that if you're someone who knows how to use developer tools to look at the API responses, you can also easily look at the API responses in many other ways.

Or you could simply create an account and see all the content on Bluesky that anyone has posted without any restrictions whatsoever, because it's all public.
